common/src/main/java/com/wynntils/mc/render/RenderUtils.java

```diff
@@ -501,6 +501,126 @@ public static void drawTooltip(List<ClientTooltipComponent> lines, PoseStack pos
         poseStack.popPose();
     }
 
+    /** drawProgressBar
+     * Draws a progress bar(textureY1 and textureY2 now specify both textures with background being on top of the bar)
+     *
+     * @param poseStack poseStack to use
+     * @param texture the texture to use
+     * @param x1 left x on screen
+     * @param y1 top y on screen
+     * @param x2 right x on screen
+     * @param y2 bottom right on screen
+     * @param textureX1 texture left x for the part
+     * @param textureY1 texture top y for the part(top of background)
+     * @param textureX2 texture right x for the part
+     * @param textureY2 texture bottom y for the part(bottom of bar)
+     * @param progress progress of the bar, 0.0f to 1.0f is left to right and 0.0f to -1.0f is right to left
+     */
+    public static void drawProgressBar(
+            PoseStack poseStack,
+            Texture texture,
+            float x1,
+            float y1,
+            float x2,
+            float y2,
+            int textureX1,
+            int textureY1,
+            int textureX2,
+            int textureY2,
+            float progress) {
+        int half = (textureY1 + textureY2) / 2;
+        drawProgressBar(poseStack, texture, x1, y1, x2, y2, textureX1, textureY1, textureX2, half + 1, progress, true);
+        drawProgressBar(poseStack, texture, x1, y1, x2, y2, textureX1, half + 1, textureX2, textureY2, progress, false);
+    }
+
+    /** drawProgressBar
+     * Draws a textured progress bar.
+     *
+     * @param poseStack poseStack to use
+     * @param texture the texture to use
+     * @param x1 left x on screen
+     * @param y1 top y on screen
+     * @param x2 right x on screen
+     * @param y2 bottom right on screen
+     * @param textureX1 texture left x for the part
+     * @param textureY1 texture top y for the part
+     * @param textureX2 texture right x for the part
+     * @param textureY2 texture bottom y for the part
+     * @param progress progress of the bar, 0.0f to 1.0f is left to right and 0.0f to -1.0f is right to left
+     * @param background is it drawing the background or not(whole ty part is background)
+     */
+    public static void drawProgressBar(
+            PoseStack poseStack,
+            Texture texture,
+            float x1,
+            float y1,
+            float x2,
+            float y2,
+            int textureX1,
+            int textureY1,
+            int textureX2,
+            int textureY2,
+            float progress,
+            boolean background) {
+        if (background) {
```

Comment:
Looks like this is basically two different functions combined, `drawProgressBarBackground` and `drawProgressBarForeground`. Maybe you should split it up instead of sending in a `backgorund` boolean?